drivers/adc/adc_stm32.c

```diff
@@ -852,6 +852,29 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 	}
 
 #elif defined(CONFIG_SOC_SERIES_STM32L4X)
+#if defined(CONFIG_SOC_STM32L412XX) || \
+	defined(CONFIG_SOC_STM32L422XX) || \
+	defined(CONFIG_SOC_STM32L432XX) || \
+	defined(CONFIG_SOC_STM32L433XX) || \
+	defined(CONFIG_SOC_STM32L452XX) || \
+	defined(CONFIG_SOC_STM32L462XX) || \
+	defined(CONFIG_SOC_STM32L4S5XX) || \
+	defined(CONFIG_SOC_STM32L4R5XX) || \
+	defined(CONFIG_SOC_STM32L4R9XX)
```

Comment:
What about `#if DT_NODE_HAS_STATUS(DT_NODELABEL(adc1), okay)` ?

Reply:
I would then go for something like?

    #if DT_NODE_HAS_STATUS(DT_NODELABEL(adc1), okay)  && DT_NODE_HAS_STATUS(DT_NODELABEL(adc3), okay)
    ...
    #elif DT_NODE_HAS_STATUS(DT_NODELABEL(adc1), okay)
    ...
    #endif

Reply:
Note that some stm32 mcus like stm32G4 or U5 have TEMP sensor on other ADC than 3: 
```
- The internal reference voltage (V REFINT ), connected to ADC1, 3, 4 and 5
- The internal temperature sensor (V TS ), connected to ADC1 and 5
```